rosenpass/src/app_server.rs

```diff
@@ -70,6 +70,51 @@ fn ipv6_any_binding() -> SocketAddr {
     SocketAddr::V6(SocketAddrV6::new(IPV6_ANY_ADDR, 0, 0, 0))
 }
 
+fn open_sockets(sockets: &mut Vec<mio::net::UdpSocket>) -> anyhow::Result<()> {
+    macro_rules! try_register_socket {
+        ($title:expr, $binding:expr) => {{
+            let r = mio::net::UdpSocket::bind($binding);
+            match r {
+                Ok(sock) => {
+                    sockets.push(sock);
+                    Some(sockets.len() - 1)
+                }
+                Err(e) => {
+                    warn!("Could not bind to {} socket: {}", $title, e);
+                    None
+                }
+            }
+        }};
+    }
+
+    let mut have_ipv4 = false;
+    let mut have_ipv6 = false;
+
+    sockets.iter().for_each(|s| {
+        let addr = s.local_addr().unwrap();
+        if addr.is_ipv4() {
+            have_ipv4 = true;
+        } else {
+            have_ipv6 = true;
+            if !s.only_v6().unwrap_or(true) && std::env::consts::OS != "macos" {
+                have_ipv4 = true;
+            }
+        }
+    });
+
+    if !have_ipv4 {
+        log::debug!("Register IPv4 socket as fallback.");
```

Comment:
If the change in behavior is accepted in rosenpass - I would promote it to `info!` and give it a descriptive text explaining why the ipv4 socket was registered.